src/audio/smart_amp/smart_amp_maxim_dsm.c

```diff
@@ -233,13 +265,40 @@ int maxim_dsm_set_param(struct smart_amp_mod_struct_t *hspk,
 	return 0;
 }
 
+int maxim_dsm_restore_param(struct smart_amp_mod_struct_t *hspk,
+			    struct comp_dev *dev)
+{
+	struct smart_amp_caldata *caldata = &hspk->param.caldata;
+	int32_t *db = (int32_t *)caldata->data;
+	int num_param = hspk->param.max_param;
+	int value[DSM_SET_PARAM_SZ_PAYLOAD];
+	enum DSM_API_MESSAGE retcode;
+	int x;
+
+	/* Restore parameter values in db to the DSM component */
+	for (x = 0 ; (x < num_param << 1) ; x++) {
+		value[DSM_SET_ID_IDX] = db[x * DSM_PARAM_MAX + DSM_PARAM_ID];
```

Comment:
I would go with i instead of x. x is not a really good name for an iterator.

Reply:
Agreed. Changed x to idx, and y to ch, respectively.